docs/rules/mapStateToProps-prefer-selectors.md

```diff
@@ -0,0 +1,52 @@
+#  Enforces that all mapStateToProps properties use selector functions. (react-redux/mapStateToProps-prefer-selectors)
+
+Using selectors in `mapStateToProps` to pull data from the store or [compute derived data](https://redux.js.org/recipes/computing-derived-data#composing-selectors) allows you to uncouple your containers from the state architecture and more easily enable memoization. This rule will ensure that every prop utilizes a selector.
+
+## Rule details
+
+The following pattern is considered incorrect:
+
+```js
+const mapStateToProps = (state) => { x: state.property }
+```
+
+```js
+connect(function(state) { 
+    return { 
+        y: state.other.property
+    }
+}, null)(App)
+```
+
+The following patterns are considered correct:
+
+```js
+const propertySelector = (state) => state.property
+const mapStateToProps = (state) => { x: propertySelector(state) }
+```
+
+```js
+const getOtherProperty = (state) => state.other.property
+connect(function(state) { 
+    return { 
+        y: getOtherProperty(state)
+    }
+}, null)(App)
+```
+
+## Rule Options
+
+```js
+...
+"react-redux/mapStateToProps-prefer-selectors": [<enabled>, {
+  "matching": <string>
+  "validateParams": <boolean>
+}]
+...
+```
+
+### `matching`
+If provided, validates the name of the selector functions against the RegExp pattern provided.
```

Comment:
would be nice to provide some examples of correct/incorrect for illustration, you have quite a few in tests.